tests/raw_idents.rs

```diff
@@ -0,0 +1,7 @@
+#[macro_use] extern crate function_name;
+
+#[named]
+fn r#if ()
+{
+    assert_eq!(function_name!(), "r#if");
+}
```

Comment:
I suspect with this impl it would not emit the `r#` part?

```suggestion
    assert_eq!(function_name!(), "if");
}
```

Reply:
I actually want the fn name correctly replicated by function_name macro, i.e. with the "r#" prefix. So the test reflects my intention.

Reply:
Gotcha. Right now the test isn't run (no `#[test]` on it), but I've fixed that, and the implementation, in #6